src/Integration.Vsix/CFamily/CLangAnalyzer.cs

```diff
@@ -102,6 +102,15 @@ public bool IsAnalysisSupported(IEnumerable<AnalysisLanguage> languages)
         }
 
         public void ExecuteAnalysis(string path,
+            IEnumerable<AnalysisLanguage> detectedLanguages,
+            IIssueConsumer consumer,
+            IAnalyzerOptions analyzerOptions,
+            IAnalysisStatusNotifier statusNotifier,
+            CancellationToken cancellationToken) =>
+            TriggerAnalysisAsync(path, detectedLanguages, consumer, analyzerOptions, statusNotifier, cancellationToken)
+                .Forget(); // fire and forget
+
+        internal /* for testing */ async Task TriggerAnalysisAsync(string path,
```

Comment:
The old code had an "internal testing" method with the same name but a different signature (a lower-level one that accepted a request).
I changed the existing tests to call this higher-level method instead (I wanted to make this method private, but there wasn't an easy way to fix the tests if I did that).

Reply:
Perhaps it's worth having the tests call this method instead so that they could await on it.
This would remove the need for `WaitForFactoryCallToComplete` and simplify the tests.